### ISSUE TITLE
about the --use-gene-bl option

### ISSUE THREAD

**Issue by juanramirez1597** [BenoitMorel/Asteroid#1]
Hello,
I just wanted to confirm if the option "--use-gene-bl" of Asteroid has been already made available. The thing is that every time I try using it, I get the error message "Unrecognized argument --use-gene-bl 
Aborting"
Thank you!
Jamie

**Comment by BenoitMorel** [comment]
Dear Jamie,
Thanks a lot for your report, this is indeed an error on my side (the parser was not able to recognize the option...). If you update your git repository and recompile, it should work fine now. 
In my experience, using gene branch lengths results (on average) in less accurate trees, so I don't necessarily recommend using this option.
Let me know if this solves your problem
Benoit

**Comment by digitalwright** [comment]
I don't think this change made it into the release builds. Would it be possible to rebuild and make available updated binaries? Thank you for considering this request.

**Comment by BenoitMorel** [comment]
Hello,
And sorry for missing the first message...
I just checked and you are right, the option is available if you build
yourself, but it is not in the release.
I'll see if I can find the time to upload a new release in the next days (I
left academia and am not active anymore)
Best,
Benoit

Le sam. 11 avr. 2026 à 03:04, digitalwright ***@***.***> a
écrit :

**Comment by BenoitMorel** [comment]
Hello again,
I've published a new release. Unfortunately I don't have access to a mac
machine, so I could only build the linux version. Let me know if that works
for you.
If someone is able to make a build for mac I would be happy to add it
Benoit

**Comment by digitalwright** [comment]
Thanks @BenoitMorel 

I get this error trying to run v1.1:
`asteroid_linux: error while loading shared libraries: libmpi_cxx.so.40: cannot open shared object file: No such file or directory`

**Comment by BenoitMorel** [comment]
Hi, thanks for the feedback
I've replaced the executable without the mpi dependencies, can you try
again?

Le sam. 11 avr. 2026 à 23:20, digitalwright ***@***.***> a
écrit :

**Comment by digitalwright** [comment]
Hi @BenoitMorel 

Sorry, I get a new error with this build:
`./asteroid_linux: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by ./asteroid_linux)
./asteroid_linux: /lib64/libc.so.6: version `GLIBC_2.38' not found (required by ./asteroid_linux)`

**Comment by BenoitMorel** [comment]
Sorry, that's because I built on a too recent architecture. I replaced the
executable, can you try again? I can't really test, but I hope it works. If
it doesn't, can you tell me your linux version?


Le dim. 12 avr. 2026 à 15:56, digitalwright ***@***.***> a
écrit :

**Comment by digitalwright** [comment]
The latest build ran without issues. I appreciate it, @BenoitMorel 

I am still getting output branch lengths of 1.0 whether I use `--use-gene-bl` or not, and Asteroid still outputs "Computing internode distances [...]" either way. Is this the expected behavior?

Separately, I tried building for Mac and there were many compiler (clang) errors. Sorry I could not figure it out...

**Comment by BenoitMorel** [comment]
Hi,
Yes this is expected, but you are right to ask. We use the branch length to
compute the distance matrix, but we don't compute the species tree branch
lengths (on purpose). The message ("computing internode distance") is a bit
misleading but I've double checked and we use the gene tree branch lengths.
Sorry for the Mac build... This is always a bit tricky and I don't think
I'll manage to get to this.
Best,
Benoit

Le sam. 18 avr. 2026 à 15:55, digitalwright ***@***.***> a
écrit :

**Comment by digitalwright** [comment]
Thank you, @BenoitMorel 

Out of curiosity, why don't you compute individual branch lengths of the species tree? I don't recall reading anything about this in the paper.

**Comment by BenoitMorel** [comment]
This is a good question. It's been a while, but I think that the reason is
that branch lengths from different gene trees are not directly comparable
(some families have higher mutation rates than others) and that I am not
sure how you would interpret those branch lengths in the end. So to avoid
any confusion we just decided not to compute them. I think there are
methods out there (for instance  Fast and accurate branch lengths
estimation for phylogenomic trees - PubMed
<https://pubmed.ncbi.nlm.nih.gov/26744021/>) for this.
In fact, that's also the reason why I usually prefer not to use this option
at all. I think we got better results when using the internode distance (on
simulated datasets)

Le sam. 18 avr. 2026 à 19:59, digitalwright ***@***.***> a
écrit :
